File: BRU-Avtopark-Avtobusov/TicketSalesApp.Core/Models/EmergencyContact.cs
// Core/Models/EmergencyContact.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
#if MODERN
using System.Text.Json.Serialization;
#elif WINDOWSXP
using Newtonsoft.Json;
#endif

namespace TicketSalesApp.Core.Models
{
    /// <summary>
    /// Emergency contact information for employees
    /// Critical for driver and field staff safety in transportation industry
    /// </summary>
    public class EmergencyContact
    {
        [Key]
        public long ContactId { get; set; }

        [Required]
        [ForeignKey("Employee")]
        public long EmployeeId { get; set; }

        [Required]
        [MaxLength(100)]
        public string ContactName { get; set; }

        [Required]
        [MaxLength(100)]
        public string Relationship { get; set; } // Spouse, Parent, Sibling, etc.

        [Required]
        [MaxLength(20)]
        public string PhoneNumber { get; set; }

        [MaxLength(20)]
        public string AlternatePhoneNumber { get; set; }

        [MaxLength(500)]
        public string Address { get; set; }

        public bool IsPrimary { get; set; } // Primary emergency contact

#if MODERN
        [JsonIgnore]
        public Employee? Employee { get; set; }
#else
        [JsonIgnore]
        public Employee Employee { get; set; }
#endif
    }
}

File: TicketSalesApp_UI_Rust_Linux/src/models/emergency_contact.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct EmergencyContact {
    pub contact_id: i64,
    pub employee_id: i64,
    pub contact_name: String,
    pub relationship: String,
    pub phone_number: String,
    pub alternate_phone_number: Option<String>,
    pub address: Option<String>,
    pub is_primary: bool,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreateEmergencyContact {
    pub employee_id: i64,
    pub contact_name: String,
    pub relationship: String,
    pub phone_number: String,
    pub alternate_phone_number: Option<String>,
    pub address: Option<String>,
    pub is_primary: bool,
}

impl EmergencyContact {
    pub fn primary_badge(&self) -> &'static str {
        if self.is_primary {
            "Основной"
        } else {
            "Дополнительный"
        }
    }
}
